src/shogun/mathematics/linalg/internal/implementation/Max.h

```diff
@@ -65,12 +67,63 @@ struct max
 	/** Scalar type */
 	typedef typename Matrix::Scalar T;
 	
-	/** Returns the largest element in a matrix or a vector */
+	/** 
+	 * Returns the largest element in a matrix or a vector.
+	 * @param m input matrix or vector
+	 * @return largest value in the input matrix or vector
+	 */
 	static T compute(Matrix m);
 };
 
+/**
+ * @brief Specialization of add for the Native backend
+ */
+template <> template <class Matrix>
+struct max<Backend::NATIVE, Matrix>
+{
+	typedef typename Matrix::Scalar T;
+
+	/**
+	 * Returns the largest element in a matrix.
+	 * @param mat input matrix
+	 * @return largest value in the input matrix
+	 */
+	static T compute(SGMatrix<T> mat)
+	{
+		return compute(mat.matrix, mat.num_cols*mat.num_rows);
+	}
+	
+	/**
+	 * Returns the largest element in a vector.
+	 * @param vec input vector
+	 * @return largest value in the input vector
+	 */
+	static T compute(SGVector<T> vec)
+	{
+		return compute(vec.vector, vec.vlen);
+	}
+
+	/**
+	 * Returns the largest element in a vector or matrix passed as a pointer.
+	 * @param m input vector or matrix
+	 * @param len length of the vector or matrix
+	 * @return largest value in the input vector or matrix
+	 */
+	static T compute(T* vec, index_t len)
+	{
+		ASSERT(len>0)
+		T maxv=vec[0];
+		for (int32_t i=1; i<len; i++)
+				maxv=std::max(vec[i], maxv);
```

Comment:
As we relying on the stl for doing the max, I suggest we just go and use `std::max_element` directly.